### PR TITLE
Made return value the actual reporter classes

### LINKED ISSUE #15: Return value for CreatePackageReport is pretty gross
Ideally, `CreatePackageReport` should generate plots AND return the network to you in a usable form.

Right now it returns a bit gross list of plotting params.

Try it:

```
thing <- CreatePackageReport('ggplot2')
str(thing, max.level = 1)
```

### PR DESCRIPTION
**Pull request by patrick-boueri**
Fixes #15 

Think it's an easy swap and a sensible return value. Plots still print.

### CODE REVIEW

**Comment by jameslamb** [comment]
I'm seeing this error on your build:

```
Error: 'roxygen2' >= 5.0.0 must be installed for this functionality
```

Can you add an `install.packages` in our Travis setup? Think it would look something like this:

```
before_install:
- Rscript -e "install.packages('roxygen2', repos = 'http://cran.rstudio.org')"
```

More docs here: https://docs.travis-ci.com/user/languages/r/

File: tests/testthat/test-CreatePackageReport.R [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+futile.logger::flog.threshold(0,name=futile.logger::flog.namespace())
+
+test_that("Test that CreatingPackageReport Runs", {
+    devtools::load_all(file.path('..','..','inst','baseballstats'))
+    #TODO: Change when generating reports
+    pdf("test_plots.pdf") #PDF doesn't actually work
+    reporters <- CreatePackageReport(packageName = "baseballstats")
+    dev.off()
+    testthat::expect_true(all(unlist(lapply(reporters,function(x) "AbstractPackageReporter" %in% class(x)))))
+    testthat::expect_true(file.exists("test_plots.pdf") && file.size("test_plots.pdf") > 0)
```
Comment (jameslamb) [inline tests/testthat/test-CreatePackageReport.R]: nice!

File: tests/testthat/test-repo_characteristics.R [hunk 2]
```diff
@@ -1,33 +1,19 @@
...
-# 
-# ##### TEST TEAR DOWN #####
-# futile.logger::flog.threshold(origLogThreshold)
-# rm(list = ls())
-# closeAllConnections()
\ No newline at end of file
+context("Checking repo characteristics")
+
+# R CMD Check stuff. Current upper lims: 0 errors, 0 warnings, 0 notes
+x <- devtools::check(pkg = file.path('..','..','.')
```
Comment (jameslamb) [inline tests/testthat/test-repo_characteristics.R]: hey we can actually totally kill this test! Travis CI runs R CMD Check already. 

Can you just remove this test entirely? Let's do it in a separate PR for cleanness

Comment (patrick-boueri) [reply tests/testthat/test-repo_characteristics.R]: done

Comment (patrick-boueri) [reply tests/testthat/test-repo_characteristics.R]: Ah shit i didn't see "Separate PR"

**Comment by patrick-boueri** [comment]
Not building though i added roxygen, also giving me a "no baseball stats" error. Any ideas?

Builds fine locally
screen shot 2017-11-21 at 5 21 13 pm

**Comment by jameslamb** [comment]
what are you trying to accomplish with `load_all`?

**Comment by patrick-boueri** [comment]
I guess i could just install_local. And also the file path could be system.file. I'll try that

**Comment by jameslamb** [comment]
ok cool. I hate `load_all`. I've never gotten it to do what it promises and it has weird interactions with `testthat`

**Pull request merged**